sale.py

```diff
@@ -276,8 +281,24 @@ def get_statement_line(self, sale):
     def transition_pay_(self):
         Sale = Pool().get('sale.sale')
 
-        active_id = Transaction().context.get('active_id', False)
-        sale = Sale(active_id)
+        sale = self.record
+        if not sale.allow_to_pay:
+            return 'end'
+
+        transaction = Transaction()
+        database = transaction.database
+        connection = transaction.connection
+
+        if database.has_select_for():
+            table = Sale.__table__()
+            query = table.select(
+                Literal(1),
+                where=(table.id == sale.id),
+                for_=For('UPDATE', nowait=True))
+            with connection.cursor() as cursor:
+                cursor.execute(*query)
+        else:
+            Sale.lock()
```

Comment:
Given that the problem occurs when user clics multiple times, I think it's better to catch the exception and ignore it (simply return), without showing a traceback to the user.

Reply:
```
psycopg2.errors.LockNotAvailable: could not obtain lock on row in relation "sale_sale"
psycopg2.errors.InFailedSqlTransaction: current transaction is aborted, commands ignored until end of transaction block
```
At the moment, abort to try/except